### PR TITLE
vm create --secrets parameter validation

### PR DESCRIPTION
**Pull request by WiseMack**
Pull #8176 without the comments

I redid the pull without using comments.

1. In line 23 The function "_validate_secrets" despite being part of azure.cli.command_modules.vm._validators  is neither imported nor used in the original file.
The function allows the validation of a secret's format, it is not among the imports and of course is not used in the _params.py file.
2. In line 172 there is a function
with self.argument_context('vm create') as c:
this vm create does NOT have a --secrets argument, I added it whole sale.
3. In line 418
There is a function
 for scope in ['vm create', 'vmss create']:
Original argument:
`c.argument('secrets', multi_ids_type, help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`', type=file_type, completer=FilesCompleter())`
New argument
`c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via "@{path}" containing [{"sourceVault":{"id":"value"},"vaultCertificates":[{"certificateUrl":"value","certificateStore":"cert store name(only on windows)"}]}]',  validator=_validate_secrets)`
Differences:
a. I removed the parametes type=file_type AND completer=FilesCompleter()) because I believe they do not apply to the --secrets parameter.
b. I added validator=_validate_secrets (The original didn't have it)
b. I removed ` and replaced them with " in "@path" and removed them altogether in {}[]

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by WiseMack** [comment]
There was a misunderstanding in #8176 where @tjprescott thought that I didn't change anything, or something, when In fact I simplydidn't understand how to use github correctly.
I added comments when It wasn't necessary to do so and confused us both. (I think)

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -415,7 +416,7 @@ def load_arguments(self, _):
             c.argument('size', help='The VM size to be created. See https://azure.microsoft.com/en-us/pricing/details/virtual-machines/ for size info.')
             c.argument('image', completer=get_urn_aliases_completion_list)
             c.argument('custom_data', help='Custom init script file or text (cloud-init, cloud-config, etc..)', completer=FilesCompleter(), type=file_type)
-            c.argument('secrets', multi_ids_type, help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`', type=file_type, completer=FilesCompleter())
+            c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via "@{path}" containing [{"sourceVault":{"id":"value"},"vaultCertificates":[{"certificateUrl":"value","certificateStore":"cert store name(only on windows)"}]}]', validator=_validate_secrets)
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: _validate_secrets shouldn't be added here but [here (process_vm_create_namespace)](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L1193), just like [here (process_vmss_create_namespace)](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L1013).

Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Also why did you remove the file_type?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -169,6 +169,7 @@ def load_arguments(self, _):
         c.argument('nsg', help='The name to use when creating a new Network Security Group (default) or referencing an existing one. Can also reference an existing NSG by ID or specify "" for none.', arg_group='Network')
         c.argument('nsg_rule', help='NSG rule to create when creating a new NSG. Defaults to open ports for allowing RDP on Windows and allowing SSH on Linux.', arg_group='Network', arg_type=get_enum_type(['RDP', 'SSH']))
         c.argument('application_security_groups', resource_type=ResourceType.MGMT_NETWORK, min_api='2017-09-01', nargs='+', options_list=['--asgs'], help='Space-separated list of existing application security groups to associate with the VM.', arg_group='Network', validator=validate_asg_names_or_ids)
+        c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via "@{path}" containing [{"sourceVault":{"id":"value"},"vaultCertificates":[{"certificateUrl":"value","certificateStore":"cert store name(only on windows)"}]}]', validator=_validate_secrets)
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: This is unnecessary.

**Comment by tjprescott** [comment]
I am closing this PR as most of the changes are unnecessary or do not accomplish what they set out to do. `_validate_secrets` was already being called in `vm create` so the changes here are not needed to make that happen. `_validate_secrets` was not being called for `vmss create` but this change would not make that happen anyways. Instead `process_vmss_create_namespace` would need to be updated.

I made that change in PR #8436.

**Pull request closed without merging**